tutorials/v7/draw_frame.cxx

```diff
@@ -0,0 +1,79 @@
+/// \file
+/// \ingroup tutorial_v7
+///
+/// \macro_code
+///
+/// \date 2020-02-20
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+/// \author Sergey Linev <s.linev@gsi.de>
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOT/RCanvas.hxx"
+#include "ROOT/RColor.hxx"
+#include "ROOT/RHistDrawable.hxx"
+#include "ROOT/RStyle.hxx"
+
+// macro must be here while cling is not capable to load
+// library automatically for outlined function see ROOT-10336
+R__LOAD_LIBRARY(libROOTHistDraw)
+
+using namespace ROOT::Experimental;
+
+auto style = std::make_shared<RStyle>(); // keep here to avoid destroy when leaving function scope
+
+void draw_frame()
+{
+   // Create the histogram.
+   RAxisConfig xaxis("x", 10, 0., 1.);
+   RAxisConfig yaxis("y", {0., 1., 2., 3., 10.});
+   auto pHist = std::make_shared<RH2D>(xaxis, yaxis);
+
+   // Fill a few points.
+   pHist->Fill({0.01, 1.02});
+   pHist->Fill({0.54, 3.02});
+   pHist->Fill({0.98, 1.02});
+   pHist->Fill({1.90, 1.02});
+   pHist->Fill({0.75, -0.02});
+
+   // Create a canvas to be displayed.
+   auto canvas = RCanvas::Create("Canvas Title");
+
+   // Make divisions
+   auto subpads = canvas->Divide(2,1);
+
+   // configure RFrame with direct API calls
+   auto frame1 = subpads[0][0]->GetOrCreateFrame();
+   frame1->AttrFill().SetColor(RColor::kBlue);
+   frame1->AttrBorder().SetColor(RColor::kRed);
+   frame1->AttrBorder().SetWidth(3);
+   frame1->Margins().SetTop(0.3_normal);
+   frame1->Margins().SetBottom(0.1_normal);
+   frame1->Margins().SetLeft(0.2_normal);
+   frame1->Margins().SetRight(0.2_normal);
+
+   frame1->AttrX().AttrLine().SetColor(RColor::kGreen);
+   frame1->AttrY().AttrLine().SetColor(RColor::kBlue);
+
+   auto draw1 = subpads[0][0]->Draw(pHist);
+   draw1->AttrLine().SetColor(RColor::kRed);
+
+   // create frame before drawing histograms
```

Comment:
That's a bit surprising - I believe it's the pad's job to ensure that the frame is first.

Reply:
In general - RFrame is just one more RDrawable in the list of primitives. And if one wants to place RText behind RFrame, such text should be before frame in list of primitives.

Reply:
The tutorial suggests that one needs to "create the frame before drawing the histogram". Why? Isn't that done automatically? What happens if the frame isn't created explicitly?

My comment is really about the lines 68, 69 in the tutorial.

Reply:
RFrame now is only container for different attributes - all optional.
If RFrame not there - default values will be used by client.

Reply:
That's a very opaque interface, not obvious at all. See how I was unable to guess that. I also don't see why I cannot have e.g. `RLine` attributes without a `RFrame`.